lib/connection.js

```diff
@@ -186,7 +199,6 @@ Connection.prototype.startup = function (callback) {
 };
 
 Connection.prototype.errorConnecting = function (err, destroy, callback) {
```

Comment:
According to the [docs](https://nodejs.org/api/net.html#net_event_error_1):
> Emitted when an error occurs. The 'close' event will be called directly following this event.

But I understand your concern, we should make sure that all situations were `errorConnecting()` is invoked, the connection is left in a consistent state. Let me check...